src/metabase/driver.clj

```diff
@@ -940,6 +940,16 @@
   dispatch-on-initialized-driver
   :hierarchy #'hierarchy)
 
+(defmulti truncate!
+  "Delete the current contents of `table-name`.
+  If something like a SQL TRUNCATE statement is supported, we use that, but may otherwise fall back to explicitly
```

Comment:
Perhaps a future refinement would be to give a default implementation in terms of a delete! driver method for non-SQL databases, and then override it as an optimization for (some?) SQL databases.